diff --git a/nbt/chunk.py b/nbt/chunk.py
index 9dc31dc..866f074 100644
--- a/nbt/chunk.py
+++ b/nbt/chunk.py
@@ -89,7 +89,7 @@ def block_id_to_name(bid):
     try:
         name = block_ids[bid]
     except KeyError:
-        name = None
+        name = 'unknown_%d' % (bid,)
         print("warning: unknown block id %i" % bid)
         print("hint: add that block to the 'block_ids' map")
     return name
@@ -252,14 +252,17 @@ class AnvilChunk(Chunk):
         try:
             version = nbt['DataVersion'].value
             if version < 1631 or version > 1631:
-                raise NotImplemented
+                #print ('Version is ', version)
+                version = 0
+                #raise NotImplemented
         except KeyError:
             version = 0
 
         # Load all sections
 
         self.sections = {}
-        for s in self.chunk_data['Sections']:
+        if 'Sections' in self.chunk_data:
+         for s in self.chunk_data['Sections']:
             self.sections[s['Y'].value] = AnvilSection(s, version)
 
 
